Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -84,14 +92,346 @@ namespace ScriptAutomation
         {
             auto func = [numSeconds]()
             {
-                auto ScriptAutomationInterface = ScriptAutomationInterface::Get();
-                auto automationComponent = azrtti_cast<ScriptAutomationSystemComponent*>(ScriptAutomationInterface);
+                auto scriptAutomationInterface = ScriptAutomationInterface::Get();
 
-                automationComponent->SetIdleSeconds(numSeconds);
+                scriptAutomationInterface->SetIdleSeconds(numSeconds);
             };
 
             ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
         }
+
+        void ResizeViewport(int width, int height)
+        {
+            auto operation = [width,height]()
+            {
+                if (Utils::SupportsResizeClientAreaOfDefaultWindow())
+                {
+                    Utils::ResizeClientArea(width, height);
+                }
+                else
+                {
+                    AZ_Error("ScriptAutomation", false, "ResizeClientArea() is not supported on this platform");
+                }
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+
+        void CapturePassTimestamp(const AZStd::string& outputFilePath)
```

Comment:
This  function can accept it's parameter by string_view or AZ::IO::PathView.
AZStd::string_view is easier to deal with for reflection, since it is already supported in the BehaviorContext.
That avoids the creation of unnecessary string when an `const char*` or character literal is supplied

```suggestion
        void CapturePassTimestamp(AZStd::string_view outputFilePath)
```

Reply:
This function is directly exposed to Lua and I'd prefer to leave it as a const string reference for now.